src/libraries/Common/src/Internal/Cryptography/HashProviderCng.cs

```diff
@@ -79,7 +81,8 @@ public override int FinalizeHashAndReset(Span<byte> destination)
                 throw Interop.BCrypt.CreateCryptographicException(ntStatus);
             }
 
-            ResetHashObject();
+            _running = false;
+            Reset();
```

Comment:
Shouldn't the order here be opposite?

Reply:
This makes the `Reset` always a no-op. Sorry for noticing it a bit too late.

Reply:
No worries; thanks, @filipnavara!

@vcsjones Let us know if this is something you can follow up on or if you need someone from the team to.

Reply:
I can follow up on this now. It would only impact Windows 7 since Windows 8+ are self resetting, but if this is indeed an issue then there is a missing test somewhere..

Reply:
@filipnavara actually I think this is working as expected. The `Reset` early return is:

```c#
if (_reusable && !_running)
    return;
```

In Windows 8+, `BCryptFinishHash` does the reset so we do not want `Reset` doing anything. In the method that you are commenting in, we call `BCryptFinishHash`. So when we call `Reset` on Windows 8+, where `_reusable` is true:

```c#
if (true && !false)  //_reusable = true; _running = false
    return;
```

So the reset does nothing. This is desirable since CNG does it for us.

In Windows 7, `_reusable` is false:

```c#
if (false && !false) //_reusable = false; _running = false
    return;
```

We skip the early return. This is desirable on Windows 7 since `BCryptFinishHash` doesn't reset.

Now, if we swapped the order, then for Windows 8+:

```c#
if (true && !true)  //_reusable = true; _running = true
    return;
```

Which means we don't do the early return, and we destroy and recreate the hash handles on Windows 8+, which is not desirable since the handles are reusable on this platform.

Reply:
You are right. Sorry for the noise 😅

Reply:
@filipnavara I certainly appreciate any eyes I can get on my pull requests. Thank you!

Reply:
Awesome; thank you both!  Good call-out on this subtlety, @filipnavara; and thanks for checking it @vcsjones.